### PR TITLE
Added notmatches operator for tagging rule

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | yes
| Translation   | yes
| Fixed tickets | #3031
| License       | MIT

I added a `notmatches` operator. 
About tests, I don't know if it's a good thing like that. @j0k3r what's your opinion?

### CODE REVIEW

File: src/Wallabag/CoreBundle/Operator/Doctrine/NotMatches.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Provides a "notmatches" operator used for tagging rules.
+ *
+ * It asserts that a given pattern is not contained in a subject, in a
+ * case-insensitive way.
+ *
+ * This operator will be used to compile tagging rules in DQL, usable
+ * by Doctrine ORM.
+ * It's registered in RulerZ using a service (wallabag.operator.doctrine.matches);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Operator/Doctrine/NotMatches.php]: `wallabag.operator.doctrine.notmatches`

File: src/Wallabag/CoreBundle/Operator/PHP/NotMatches.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Provides a "notmatches" operator used for tagging rules.
+ *
+ * It asserts that a given pattern is not contained in a subject, in a
+ * case-insensitive way.
+ *
+ * This operator will be used to compile tagging rules in PHP, usable
+ * directly on Entry objects for instance.
+ * It's registered in RulerZ using a service (wallabag.operator.array.matches);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Operator/PHP/NotMatches.php]: `wallabag.operator.array.notmatches`

**Comment by j0k3r** [comment]
About test, run the coverage and check that both added files are covered :wink:

**Comment by nicosomb** [comment]
capture d ecran 2017-04-25 a 16 01 30

**Pull request merged**